src/deploy.sh

```diff
@@ -184,87 +186,68 @@ function deploy_parser_options()
     options_values['REBOOT']=1
   fi
 
-  IFS=' ' read -r -a options <<< "$raw_options"
-  for option in "${options[@]}"; do
-    if [[ "$option" =~ ^(--.*|-.*|test_mode) ]]; then
-      if [[ "$enable_collect_param" == 1 ]]; then
-        options_values['ERROR']='expected paramater'
-        return 22
-      fi
+  eval "set -- $options"
 
-      case "$option" in
-        --remote)
-          options_values['TARGET']="$REMOTE_TARGET"
-          continue
-          ;;
-        --local)
-          options_values['TARGET']="$LOCAL_TARGET"
-          continue
-          ;;
-        --vm)
-          options_values['TARGET']="$VM_TARGET"
-          continue
-          ;;
-        --reboot | -r)
-          options_values['REBOOT']=1
-          continue
-          ;;
-        --modules | -m)
-          options_values['MODULES']=1
-          continue
-          ;;
-        --list | -l)
-          options_values['LS']=1
-          continue
-          ;;
-        --ls-line | -s)
-          options_values['LS_LINE']=1
-          continue
-          ;;
-        --uninstall | -u)
-          enable_collect_param=1
-          uninstall=1
-          continue
-          ;;
-        test_mode)
-          options_values['TEST_MODE']='TEST_MODE'
-          ;;
-        *)
-          options_values['ERROR']="$option"
-          return 22 # EINVAL
-          ;;
-      esac
-    else # Handle potential parameters
-      if [[ "$uninstall" != 1 &&
-        ${options_values['TARGET']} == "$REMOTE_TARGET" ]]; then
-        populate_remote_info "$option"
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      --remote)
+        options_values['TARGET']="$REMOTE_TARGET"
+        populate_remote_info "$2"
```

Comment:
Now that you changed this `--remote` behavior to enforce a parameter, we need to update the `help` and the documentation.